client/lib/posts/utils.js

```diff
@@ -179,33 +190,23 @@ export const getPermalinkBasePath = function( post ) {
 	let path = post.URL;
 
 	// if we have a permalink_URL, utlize that
-	if ( ! this.isPublished( post ) && post.other_URLs && post.other_URLs.permalink_URL ) {
+	if ( ! isPublished( post ) && post.other_URLs && post.other_URLs.permalink_URL ) {
 		path = post.other_URLs.permalink_URL;
 	}
 
-	return this.removeSlug( path );
+	return removeSlug( path );
 };
 
 export const getPagePath = function( post ) {
 	if ( ! post ) {
 		return;
 	}
-	if ( ! this.isPublished( post ) ) {
-		return this.getPermalinkBasePath( post );
+	if ( ! isPublished( post ) ) {
+		return getPermalinkBasePath( post );
 	}
 
-	return this.removeSlug( post.URL );
-};
-
-export const removeSlug = function( path ) {
-	if ( ! path ) {
-		return;
-	}
-
-	const pathParts = path.slice( 0, -1 ).split( '/' );
-	pathParts[ pathParts.length - 1 ] = '';
-
-	return pathParts.join( '/' );
+	const postUrl = get( post, [ 'other_URLs', 'permalink_URL' ] ) || post.URL;
+	return removeSlug( postUrl );
```

Comment:
... and you can also revert this back to just `return removeSlug( post.URL )` -- the permalink case was already handled.

Reply:
While I think thats true, it'll fail one of the unit tests.

Reply:
edit: i lied. it works fine now that the `return` was added back :)